src/query/sql/src/planner/binder/window.rs

```diff
@@ -0,0 +1,379 @@
+                        args,
+                        params,
+                        window,
+                    } => {
+                        if let Some(window) = window {
+                            window_order_by.extend_from_slice(&window.order_by);
+                        }
+                    }
+                    _ => continue,
+                },
+            }
+        }
+        window_order_by
+    }
+
+    pub(super) async fn fetch_window_order_items(
+        &mut self,
+        from_context: &BindContext,
+        scalar_items: &mut HashMap<IndexType, ScalarItem>,
+        projections: &[ColumnBinding],
+        window_order_by: &[OrderByExpr],
+        distinct: bool,
+    ) -> Result<OrderItems> {
+        let mut order_items = Vec::with_capacity(window_order_by.len());
+        for order in window_order_by {
+            match &order.expr {
+                Expr::ColumnRef {
+                    database: ref database_name,
+                    table: ref table_name,
+                    column: ref ident,
+                    ..
+                } => {
+                    // We first search the identifier in select list
+                    let mut found = false;
+                    let database = database_name
+                        .as_ref()
+                        .map(|ident| normalize_identifier(ident, &self.name_resolution_ctx).name);
+                    let table = table_name
+                        .as_ref()
+                        .map(|ident| normalize_identifier(ident, &self.name_resolution_ctx).name);
+                    let column = normalize_identifier(ident, &self.name_resolution_ctx).name;
+
+                    for item in projections.iter() {
+                        if BindContext::match_column_binding(
+                            database.as_deref(),
+                            table.as_deref(),
+                            column.as_str(),
+                            item,
+                        ) {
+                            order_items.push(OrderItem {
+                                expr: order.clone(),
+                                index: item.index,
+                                name: item.column_name.clone(),
+                                need_eval_scalar: scalar_items.get(&item.index).map_or(
+                                    false,
+                                    |scalar_item| {
+                                        !matches!(
+                                            &scalar_item.scalar,
+                                            ScalarExpr::BoundColumnRef(_)
+                                        )
+                                    },
+                                ),
+                            });
+                            found = true;
+                            break;
+                        }
+                    }
+
+                    if found {
+                        continue;
+                    }

```

Comment:
I think we don't need to do this for `order by` in window functions. It's different from `order by` clause in the query body, we can just throw an error if no matched column is found.